### PR TITLE
fix torch error for bincount

### PR DESCRIPTION
**Pull request by divyashreepathihalli**

### CODE REVIEW

File: keras_core/layers/preprocessing/discretization_test.py [hunk 1]
```diff
@@ -64,9 +64,9 @@ def test_correctness(self):
         layer = layers.Discretization(
             bin_boundaries=[0.0, 0.5, 1.0], output_mode="count"
         )
-        output = layer(np.array([[0.1, 0.8, 0.9]]))
+        output = layer(np.array([0.1, 0.8, 0.9]))
         self.assertTrue(backend.is_tensor(output))
-        self.assertAllClose(output, np.array([[0, 1, 2, 0]]))
```
Comment (fchollet) [inline keras_core/layers/preprocessing/discretization_test.py:69]: The layer is expected to work for 2D inputs like these. We need to fix this case too.

File: keras_core/utils/numerical_utils.py [hunk 2]
```diff
@@ -139,16 +139,16 @@ def encode_categorical_inputs(
...
-        one_hot_input = backend_module.nn.one_hot(inputs, depth)
-        bincounts = backend_module.numpy.where(
-            backend_module.numpy.any(one_hot_input, axis=-2), 1, 0
-        )
+      one_hot_input = backend_module.nn.one_hot(inputs, depth)
+      bincounts = backend_module.numpy.where(
+          backend_module.numpy.any(one_hot_input, axis=-2), 1, 0
+      )
+    else:
+      bincounts = backend_module.numpy.bincount(
```
Comment (fchollet) [inline keras_core/utils/numerical_utils.py:148]: Use 4-space indent.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/layers/preprocessing/discretization_test.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_core/layers/preprocessing/discretization_test.py]: done!

File: keras_core/utils/numerical_utils.py (hunk 2 above)
Comment (divyashreepathihalli) [reply keras_core/utils/numerical_utils.py]: done!

**Review by fchollet — COMMENTED** [review]
Thanks for the update!

File: keras_core/backend/jax/numpy.py [hunk 3]
```diff
@@ -13,10 +13,13 @@ def add(x1, x2):
 
 def bincount(x, weights=None, minlength=0):
     if len(x.shape) == 2:
-        bincounts = [
-            jnp.bincount(arr, weights=weights, minlength=minlength)
-            for arr in list(x)
-        ]
+        if weights is None:
+            bincounts = [jnp.bincount(arr, minlength=minlength) for arr in x]
```
[the remark this answers is not in the document]
Comment (divyashreepathihalli) [reply keras_core/backend/jax/numpy.py:17]: It is actually not working, I was confused why those tests passed previously. But I have updated the tests and they do fail when the input is 2d now. [`jax.numpy.bincount`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.bincount.html) only supports 1d arrays 
updated the implementation to use map and def

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**